src/fastfetch.c

```diff
@@ -1471,6 +1485,13 @@ int main(int argc, const char** argv)
     if(data.structure.length == 0)
         ffStrbufAppendS(&data.structure, FASTFETCH_DATATEXT_STRUCTURE);
 
+    if(ffStrbufContainIgnCaseS(&data.structure, ":cpuusage:"))
+        ffPrepareCPUUsage();
+    if(ffStrbufContainIgnCaseS(&data.structure, ":publicip:"))
+        ffPreparePublicIp(&instance);
+    if(ffStrbufContainIgnCaseS(&data.structure, ":weather:"))
+        ffPrepareWeather(&instance);
+
```

Comment:
It should work without the `:` too, this way modules at the start and the end of the structure can be matched too.